example-get-started-experiments/code/src/inference.py

```diff
@@ -0,0 +1,46 @@
+from pathlib import Path
+from typing import Optional
+
+import cv2
+import numpy as np
+from fire import Fire
+from sagemaker.deserializers import JSONDeserializer
+from sagemaker.pytorch import PyTorchPredictor
+from ultralytics import YOLO
+
+
+def inference(
+    img_path: str,
+    endpoint_name: Optional[str] = None,
+    model_path: Optional[str] = None,
+    output_path: str = "predictions",
+):
+
+    if endpoint_name is not None and model_path is not None:
```

Comment:
why do we need two inferences (two python scripts) ... both of them something related to endpoints

Reply:
(`sagemaker/code/inference.py`) is bundled alongside the model file and uploaded to Sagemaker, it can't be used from a user perspective. 
It includes methods following conventions required by Sagemaker.